### ISSUE TITLE
Add a buffer before transactional producer

### ISSUE THREAD

**Issue by seglo** [akka/alpakka-kafka#1282]
### Short description

Add a buffering stage before the transactional producer so that some amount of messages can still be processed by user business logic while a commit is occurring in the producer stage (and back-pressuring).

**Comment by jrudolph** [comment]
Maybe it would be enough to document the implications of the transactional behavior on the user stream, instead? And add a snippet that people can experiment with a buffer themselves?

**Comment by seglo** [comment]
Good idea. We have enough of a maintenance burden with transactions as it is.

**Comment by ennru** [comment]
Let's try it in the benchmarks. (not that they'd have a lot of business logic, ofc)
